pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/region/provisionpolicy/policy_rama.go

```diff
@@ -79,19 +82,44 @@ func (p *PolicyRama) Update() error {
 	// pass the current value to regulator to avoid slow start from zero after restart
 	p.regulator.SetLatestCPURequirement(int(cpuSize))
 
-	cpuSizeAdjustedGlobal := math.Inf(-1)
+	cpuAdjustedRaw := math.Inf(-1)
 
 	// run pid control for each indicator
 	for _, metricName := range p.configuredIndicatorMetrics {
 		indicator := p.Indicators[metricName]
 		controller := p.controllers[metricName]
 
-		cpuSizeAdjusted := controller.Adjust(cpuSize, indicator.Target, indicator.Current)
-		if cpuSizeAdjusted > cpuSizeAdjustedGlobal {
-			cpuSizeAdjustedGlobal = cpuSizeAdjusted
+		controller.SetEssentials(p.ResourceEssentials)
+		cpuAdjusted := controller.Adjust(cpuSize, indicator.Target, indicator.Current)
+
+		if cpuAdjusted > cpuAdjustedRaw {
+			cpuAdjustedRaw = cpuAdjusted
+		}
+	}
+	cpuAdjustedRestricted := cpuAdjustedRaw
+
+	// restrict cpu size adjusted
+	if p.ControlEssentials.IsReclaimOverlap {
+		reclaimedUsage, reclaimedCnt := p.getReclaimStatus()
+		klog.Infof("[qosaware-cpu-rama] reclaim usage %.2f #container %v", reclaimedUsage, reclaimedCnt)
+
+		reason := ""
+		if reclaimedCnt <= 0 {
+			// do not reclaim if no reclaimed containers
+			cpuAdjustedRestricted = p.ResourceUpperBound
+			reason = "no reclaimed container"
+		} else {
+			// do not overlap more if reclaim usage is below threshold
+			threshold := p.ResourceUpperBound - reclaimedUsage - types.ReclaimUsageMarginForOverlap
```

Comment:
why we need to minus types.ReclaimUsageMarginForOverlap here?

Reply:
Minus a constant so that reclaimed containers may have change to use more cpus. We do this here to avoid potentially overlap too many cpus with reclaimed containers when current reclaim usage is low.

Reply:
Minus a constant so that reclaimed containers may have chance to use more cpus. We do this here to avoid potentially overlap too many cpus with reclaimed containers when current reclaim usage is low and indicator always below target.

Reply:
okay, make sense